zerver/lib/actions.py

```diff
@@ -4631,13 +4633,45 @@ def get_service_dicts_for_bots(user_profile_id: str) -> List[Dict[str, Any]]:
             pass
     return service_dicts
 
+def get_service_dicts_for_bots(user_profile_ids: List[str]) -> Dict[str, List[Dict[str, Any]]]:
+    bot_profiles = get_user_profiles_by_ids(user_profile_ids)
```

Comment:
I think you can save this database query by passing the list of `user_profile` objects into this function - the caller has it already.